### ISSUE TITLE
default_url_options is ignored in console

### ISSUE THREAD

**Issue by grosser** [rails/rails#43357]
## What Ruby, Rails and RSpec versions are you using?

Ruby version: 2.7.2
Rails version: 6.1.4.1

## Observed behaviour

```
routes.default_url_options.merge!(host: 'localhost:3000', protocol: 'http') # development.rb

rails c
app.get '/'
-> 403
app.response
"Blocked host: www.example.com"
```

## Expected behaviour

using localhost:3000

## Can you provide an example app?

simple `rails new` reproduces it

## Root cause

https://github.com/rails/rails/blob/main/railties/lib/rails/console/app.rb#L13 which should be using default_url_options

**Comment by DerekCrosson** [comment]
I'm happy to have a look at this if I'm allowed to. I'm trying to make contributions to open source projects but doing at least one a month

**Comment by DerekCrosson** [comment]
> 👋 Trying to help triage old issues. The `default_url_options` on the route set is supposed to take only effect when calling a url helper. In your case you are passing a hardcoded path "/".
> 
> `app.get(root_url)` correctly makes use of the `default_url_options`.
> 
> I added the "More info needed" label in case you or someone else would like to comment, otherwise the issue will be closed automatically in two weeks.
> 
> Thanks for reporting the issue and sorry for the "delayed" answer 😅

@Edouard-chin I'm not sure if this is still an issue? If it is I'm happy to fix it

**Comment by Edouard-chin** [comment]
I don't think this ever was an issue. It's supposed to work like that when you don't use a url helper

**Comment by grosser** [comment]
little unexpected that it is supposed to be that way, but thx for the reply!

**Comment by grosser** [comment]
FYI fix for rails 8+
```
# https://github.com/rails/rails/issues/43357
# app.url_for Project -> "http://localhost:3000/projects"
Rails.application.console do
  require 'rails/commands/console/irb_console'
  Rails::Console::AppInstance.prepend(Module.new do
    def execute(*)
      super.tap { |s| s.host! "localhost" }
    end
  end)
end
```
